dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+                        Logger.warn(classToUse, "Property " + property + "not set for " + classToUse.getName());
+                    }
+                }
+            }
+
+            ret.add(object);
+        }
+        return ret;
+    }
+
+    /**
+     * Searches a property recursively through classes and superclasses
+     * @param classToUse
+     * @param fieldType
+     * @param property
+     * @return
+     * @throws NoSuchFieldException
+     */
+    private static boolean isFieldPresent(final Class classToUse, Class fieldType, String property)
+            throws NoSuchFieldException {
+
+        try{
+            return classToUse.getDeclaredField(property).getType() == fieldType;
+        }catch(NoSuchFieldException e){
+            if (classToUse.getSuperclass()!=null) {
+                return isFieldPresent(classToUse.getSuperclass(), fieldType, property);
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Link> convertDotConnectMapToLink(final List<Map<String,String>> results)
+            throws ParseException {
+
+        List<Link> ret;
+
+        ret = new ArrayList<>();
+
+        if(results != null && !results.isEmpty()){
+            for (Map<String, String> map : results) {
+                final Link link = new Link();
+                link.setInode(map.get("inode"));
+
+                if (map.get("show_on_menu") != null && !map.get("show_on_menu").isEmpty()){
+                    link.setShowOnMenu(Boolean.parseBoolean(map.get("show_on_menu")));
+                }
+
+                link.setTitle(map.get("title"));
+
+                if (map.get("mod_date") != null && !map.get("mod_date").isEmpty()){
+                    link.setModDate(df.parse(map.get("mod_date")));
+                }
+
+                link.setModUser(map.get("mod_user"));
+
+                if (map.get("sort_order") != null && !map.get("sort_order").isEmpty()){
+                    link.setSortOrder(Integer.parseInt(map.get("sort_order")));
+                }
+
+                link.setFriendlyName(map.get("friendly_name"));
+
+                link.setIdentifier(map.get("identifier"));
+
+                link.setProtocal(map.get("protocal"));
+
+                link.setUrl(map.get("url"));
+
+                link.setTarget(map.get("target"));
+
+                link.setInternalLinkIdentifier(map.get("internal_link_identifier"));
+
+                link.setLinkType(map.get("link_type"));
+
+                link.setLinkCode(map.get("link_code"));
+
+                ret.add(link);
+            }
+        }
+
+
+        return ret;
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Identifier> convertDotConnectMapToIdentifier(final List<Map<String,String>> results)
+            throws ParseException {
+        final List<Identifier> ret = new ArrayList<>();
+        if(results == null || results.size()==0){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)